File: atividades_verde/C/palindromo_recursivo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int isPalindromo(char palavra[], int i, int y) {
    int resposta = 0;
    if(i < strlen(palavra) / 2) {
        if(palavra[i] != palavra[y]) {
            resposta = 1;
            resposta = resposta + isPalindromo(palavra, y, i);
        }
        resposta = resposta + isPalindromo(palavra, i + 1, y - 1);
    }

    return resposta;
}

bool eIgual(char x[], char y[]) {
    bool resposta = false;
    if(strlen(x) == strlen(y)) {
        resposta = true;
        for(int i = 0; i < strlen(x); i++) {
            if(x[i] != y[i]) {
                resposta = false;
                i = strlen(x);
            }
        }
    }
    return resposta;
}

int main() {
    char palavra[500];
    scanf("%[^\n]", palavra);
    getchar();
    while(!eIgual(palavra, "FIM")) {
        int resposta = isPalindromo(palavra, 0, strlen(palavra) - 1);
        if(resposta >= 1) 
            printf("NAO\n");
        else    
            printf("SIM\n");
        scanf("%[^\n]", palavra);
        getchar();
    }
    return 0;
}

File: atividades_verde/JAVA/Palindromo_Recursivo.java
public class Palindromo_Recursivo {
    public static int isPalindromo(String x, int i, int y) { // Função para determinar se é palindromo recursivamente.
        int resposta = 0;
        int tam = x.length();
        if(i < tam/2) {
            if(x.charAt(i) != x.charAt(y)) {
                resposta = 1;
                isPalindromo(x, y, i);
            } else {
                resposta = resposta + isPalindromo(x, i + 1, y - 1);
            }
        }
        return resposta;
    }

    public static boolean eIgual(String x, String y) { // Função para comparar duas strings
        boolean resposta = false;
        int tam1 = x.length(), tam2 = y.length();
        if(tam2 == tam1) {
            resposta = true;
            for(int i = 0; i < tam2; i++) {
                if(x.charAt(i) != y.charAt(i)) {
                    resposta = false;
                    i = tam1;   
                }
            }
        }
        return resposta;
    }
    public static void main(String[] args) {
        String palavra = MyIO.readLine();
        while(!eIgual(palavra, "FIM")) {
            int resposta = isPalindromo(palavra, 0, palavra.length() - 1);
            if(resposta >= 1) 
                MyIO.println("NAO");
            else 
                MyIO.println("SIM");
            palavra = MyIO.readLine();
        }
    }   
}
